### PR TITLE
[5.x] Cassiopeia editor.scss

### PR DESCRIPTION
**Pull request by brianteeman**
Tinymce in its current implementation is not a true wysiwyg editor for the cassiopeia template as it is not aware of the cassiopeia css. This can severely restrict what people think is possible with tinmyce and creates confusion that what they edit doesn't match what is displayed.

By default tinymce will look for the presence of an editor.css file in the site template and load the classes from that. Prior to this PR the only thing in that file was some internal classes. This PR creates a true editor.css file for the cassiopeia template. It is generated by the editor.scss file which is a subset of the template.scss which keeps the size down.

One downside of adding this editor.css is that tinymce in its current configuration will display all the classes in one long list on the formats menu/button. Therefore I have changed the default behaviour to NOT display the list (the css is still loaded) and added a new configuration option (default is off) that will let you turn it on if you want. This means that there is a small b/c break so documentation will be created.

###  Testing
As this is an scss change you will need to either `run npm build:css` or use one of the pre built packages.
The eaisest thing to check would be to edit the typography article of the sample data.

### Before PR

image

also check the Display-1 text at the bottom of the file

### After PR

image

## Frontend
This pr works just the same on the frontend where _perhaps_ it is a more obvious improvement.

### Before PR

https://github.com/joomla/joomla-cms/assets/1296369/6291f940-c92b-4300-9d62-4178ed1b7dc8

### After PR

https://github.com/joomla/joomla-cms/assets/1296369/c2ecdff4-3446-4689-aaef-946c2af0517b

Please select:
- [x] Documentation link for docs.joomla.org: https://docs.joomla.org/Chunk4x:Extensions_Plugin_Manager_Edit_Editor_Group/en will be updated once merged
- [ ] No documentation changes for docs.joomla.org needed

- [x] Pull Request link for manual.joomla.org: [<link>](https://github.com/joomla/Manual/pull/129)
- [ ] No documentation changes for manual.joomla.org needed

### CODE REVIEW

File: administrator/language/en-GB/plg_editors_tinymce.ini [hunk 1]
```diff
@@ -30,6 +30,8 @@ PLG_TINY_FIELD_CUSTOM_CONTENT_TEMPLATE_PATH_DESC="A directory named \"tinymce\"
 PLG_TINY_FIELD_CUSTOM_CONTENT_TEMPLATE_PATH_LABEL="Content Template Directory"
 PLG_TINY_FIELD_CUSTOM_CSS_DESC="Optional CSS file that will override the standard editor.css file. Enter a file name to point to a file in the CSS folder of the default template (for example, media/templates/site/cassiopeia/css/). Or enter a full URL path to the custom CSS file. If you enter a value in this field, this file will be used instead of the editor.css file."
 PLG_TINY_FIELD_CUSTOM_CSS_LABEL="Custom CSS Classes"
+PLG_TINY_FIELD_CUSTOM_CSS_MENU_DESC="By deafult CSS classes available from the template or custom css are not shown in the formats menu dropdown. If enabled they will be displayed in a long list."
```
Comment (richard67) [inline administrator/language/en-GB/plg_editors_tinymce.ini]:
```suggestion
PLG_TINY_FIELD_CUSTOM_CSS_MENU_DESC="By default CSS classes available from the template or custom css are not shown in the formats menu dropdown. If enabled they will be displayed in a long list."
```

**Comment by richard67** [comment]
@brianteeman Can it be that the before and after PR videos for the frontend are mixed up?

**Comment by brianteeman** [comment]
> @brianteeman Can it be that the before and after PR videos for the frontend are mixed up?

they were indeed. I have corrected the posrt - thanks

**Comment by brianteeman** [comment]
because if you dont then you dont have a wyiwyg editor

**Comment by HLeithner** [comment]
Yes but what has changed between https://github.com/joomla/joomla-cms/pull/31828#issuecomment-753537194 and now?

Your quote:
> There is absolutely no way that an editor.css should be over 1000 lines!!

**Comment by brianteeman** [comment]
Because
1. I was wrong as I was saying it should be in a system editor.css and a small cassiopeia edirtor.css override. But now that we are being stricter about system/core markup not being bootstrap it is not possible.
2. That PR didnt implement the correct classes for the cassiopeia template - it was just native bootstrap
3. That PR had the problem of the dropdowns which I resolved here

**Comment by dgrammatiko** [comment]
Couple of things here:
- you could have an `@import` statement at the beginning of the file pointing to the template.css (so the editor has only the TinyMCE related code, easier to edit)
- This approach is not realistic EVEN for cassiopeia. Let me explain: Most templates (also Cassiopeia) use some of the params to customise some colours, fonts, etc. By adding a static file these params cannot be emulated correctly. The only way to do this **correctly** is to have PHP create the editor.css, similar to what I'm doing with https://github.com/dgrammatiko/wysiwyg-tinymce-j4.1
- Last comment, if the RL of 4.1 and later releases weren't against to my proposal for a similar patch to com_ajax as the one for plugins and modules the url could have been part of the Helper file but right now (unless someone is willing to fix that and PD accept it) only calling a normal route with a different php entry point (ie I 'm using editor.php) is possible.

In short, this is not a complete solution, and once again the claim WYSIWYG is by approximation...

**Comment by brianteeman** [comment]
Could you explain the first point please. My understanding is that if you simply import the entire template css then the editor has all the css. The intention here was to only include the minimum

**Comment by dgrammatiko** [comment]
> My understanding is that if you simply import the entire template css then the editor has all the css. 

Nope, that not what's happening in reality. You hard code:
- The fonts: https://github.com/joomla/joomla-cms/blob/a8be6ef925e601fb7e3f38c5dda7b4ddbbc85f3a/templates/cassiopeia/index.php#L48-L65
- The inline color parts: https://github.com/joomla/joomla-cms/blob/a8be6ef925e601fb7e3f38c5dda7b4ddbbc85f3a/templates/cassiopeia/index.php#L72-L80
- The extra color CSS: https://github.com/joomla/joomla-cms/blob/a8be6ef925e601fb7e3f38c5dda7b4ddbbc85f3a/templates/cassiopeia/index.php#L40-L42

**Comment by brianteeman** [comment]
but then you are still importing hundreds of lines of css that are not relevant to the editor. That is what I was aiming to avoid

**Comment by brianteeman** [comment]
Am I being really daft or is the inline style for the color complletely wrong. There is no way in cassiopeia to set those colours. That is a functionality only present in Atum

**Comment by brianteeman** [comment]
Thanks to others for pointing out when those inline color styles are used ie when using the media manager in the frontend. So definitely not needed in the editor.css

**Comment by dgrammatiko** [comment]
> Thanks to others for pointing out when those inline color styles are used ie when using the media manager in the frontend. So definitely not needed in the editor.css

Still, this approach is completely ignorant of the colours, fonts:

- The extra color CSS: https://github.com/joomla/joomla-cms/blob/a8be6ef925e601fb7e3f38c5dda7b4ddbbc85f3a/templates/cassiopeia/index.php#L40-L42


But most importantly, for me, it's a monkey patch if you try to emulate the styles ignoring that almost every template out there has some params that control aspects of the styling and therefore can only be done using PHP.

**Comment by brianteeman** [comment]
This is only for cassiopeia not for any other template!
